docs/book/src/plugins/extending/extending_cli_features_and_plugins.md

```diff
@@ -0,0 +1,411 @@
+# Extending CLI Features and Plugins
+
+Kubebuilder provides an extensible architecture to scaffold
+projects using plugins. These plugins allow you to customize the CLI
+behavior or integrate new features.
+
+In this guide, we’ll explore how to extend CLI features,
+create custom plugins, and bundle multiple plugins.
+
+## Creating Custom Plugins
+
+To create a custom plugin, you need to implement
+the [Kubebuilder Plugin interface][plugin-interface].
+
+This interface allows your plugin to hook into Kubebuilder’s
+commands (`init`, `create api`, `create webhook`, etc.)
+and add custom logic.
+
+### Example of a Custom Plugin
+
+You can create a plugin that generates both
+language-specific scaffolds and the necessary configuration files,
+using the [Bundle Plugin](#bundle-plugin). This example shows how to
+combine the Golang plugin with a Kustomize plugin:
+
+```go
+import (
+    kustomizecommonv2 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v2"
+    golangv4 "sigs.k8s.io/kubebuilder/v4/pkg/plugins/golang/v4"
```

Comment:
this import is not used in example, should it be removed?

Reply:
I replaced all across the project.